src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -49,20 +51,49 @@ public partial class RedisCache : IDistributedCache, IDisposable
     private const string AbsoluteExpirationKey = "absexp";
     private const string SlidingExpirationKey = "sldexp";
     private const string DataKey = "data";
+
+    // combined keys - same hash keys fetched constantly; avoid allocating an array each time
+    private static readonly RedisValue[] HashMembers_AbsoluteExpiration_SlidingExpiration_Data = new RedisValue[] { AbsoluteExpirationKey, SlidingExpirationKey, DataKey };
+    private static readonly RedisValue[] HashMembers_AbsoluteExpiration_SlidingExpiration = new RedisValue[] { AbsoluteExpirationKey, SlidingExpirationKey };
+
+    private static RedisValue[] GetHashFields(bool getData) => getData
+        ? HashMembers_AbsoluteExpiration_SlidingExpiration_Data
+        : HashMembers_AbsoluteExpiration_SlidingExpiration;
+
     private const long NotPresent = -1;
     private static readonly Version ServerVersionWithExtendedSetCommand = new Version(4, 0, 0);
 
-    private volatile IConnectionMultiplexer? _connection;
-    private IDatabase? _cache;
+    private volatile IDatabase? _cache;
     private bool _disposed;
     private string _setScript = SetScript;
 
     private readonly RedisCacheOptions _options;
-    private readonly string _instance;
+    private readonly RedisKey _instancePrefix;
     private readonly ILogger _logger;
 
     private readonly SemaphoreSlim _connectionLock = new SemaphoreSlim(initialCount: 1, maxCount: 1);
 
+    private long _lastConnectTicks = DateTimeOffset.UtcNow.Ticks;
+    private long _firstErrorTimeTicks;
+    private long _previousErrorTimeTicks;
+
+    //// StackExchange.Redis will also be trying to reconnect internally,
+    //// so limit how often we recreate the ConnectionMultiplexer instance
+    //// in an attempt to reconnect
+    private readonly TimeSpan ReconnectMinInterval = TimeSpan.FromSeconds(60);
+    private readonly TimeSpan ReconnectErrorThreshold = TimeSpan.FromSeconds(30);
```

Comment:
```suggestion
    // Never reconnect within 60 seconds of the last attempt to connect or reconnect.
    private readonly TimeSpan ReconnectMinInterval = TimeSpan.FromSeconds(60);
    // Only reconnect if errors have occurred for at least the last 30 seconds.
    // This count resets if there are no errors for 30 seconds
    private readonly TimeSpan ReconnectErrorThreshold = TimeSpan.FromSeconds(30);
```


Am I understanding this correctly?

Also, do we want to make `ReconnectMinInterval` and `ReconnectErrorThreshold` configurable? How about an option to disable "force reconnect" altogether?